File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool func(string &s,int i,int j)
    {
        if(i>=j)
        {
            cout<<i<<" "<<j;
            return true;
        }
        if(s[i]!=s[j])return false;
        cout<<i<<" "<<j;
        return func(s,++i,--j);
        
    }
    bool isPalindrome(string s) {
        transform(s.begin(),s.end(),s.begin(),::tolower);
        
       

size_t i = 0;
size_t len = s.length();
while(i < len){
    if (!isalnum(s[i]) || s[i] == ' '){
        s.erase(i,1);
        len--;
    }else
        i++;
}
        cout<<s;
        bool res = func(s,0,s.length()-1);
        return res;
    }
};

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String str) {
        str = str.toLowerCase();

		str = str.replaceAll("[^a-zA-Z0-9]","");
		int p1=0;
		int p2=str.length()-1;
        while(p1<=p2)
		{
			if(str.charAt(p1++)!=str.charAt(p2--))
			{
				return false;
			}
		}
		return true;
    }
}